Docs:Adjust azuread.md for Powershell UUID command

RobinJoe opened the pull request:
<!--

Thank you for sending a pull request! Here are some tips:

1. If this is your first time, please read our contribution guide at https://github.com/grafana/grafana/blob/master/CONTRIBUTING.md

2. Ensure you include and run the appropriate tests as part of your Pull Request.

3. In a new feature or configuration option, an update to the documentation is necessary. Everything related to the documentation is under the docs folder in the root of the repository.

4. If the Pull Request is a work in progress, make use of GitHub's "Draft PR" feature and mark it as such.

5. If you can not merge your Pull Request due to a merge conflict, Rebase it. This gets it in sync with the master branch.

6. Name your PR as "<FeatureArea>: Describe your change", e.g. Alerting: Prevent race condition. If it's a fix or feature relevant for the changelog describe the user impact in the title. The PR title is used to auto-generate the changelog for issues marked with the "add to changelog" label.

-->

**What this PR does / why we need it**:

-Updates the docs to include **more Unique ID information for Windows users**. This is intended to improve user experience.
-Adjusts passive to active voice where it improves the documentation.
-Checking for style guide compliance.

**Which issue(s) this PR fixes**:

<!--

* Automatically closes linked issue when the Pull Request is merged.

Usage: "Fixes #<issue number>", or "Fixes (paste link of issue)"

-->

Fixes #23094

**Special notes for your reviewer**:

For the configuration key and value pairs, does each role need its own unique ID? Or, can users enter the same UUID they generate for each role? If each role needs its own ID, I can include a few words to explain that need.

RobinJoe committed:
Docs:Adjust azuread.md for Powershell UUID command

Why this change is needed:

-Update the docs to include more Unique ID information for Windows users. This is intended to improve user experience.
-Adjust passive to active voice where it improves the documentation.
-Checking for style guide compliance.

Closes issue: #23094

achatterjee-grafana reviewed (commented):
Typo "possibile" needs to be fixed.

achatterjee-grafana committed:
Update azuread.md

Fixed typo "possibile".

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/grafana?pullRequest=27323) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/grafana/grafana?pullRequest=27323) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: RobinJoe<br/>:x: achatterjee-grafana<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/grafana/grafana?pullRequest=27323) it.</sub>

oddlittlebird commented on docs/sources/auth/azuread.md:
```diff
@@ -14,7 +14,7 @@ weight = 3
 > Only available in Grafana v6.7+
 
-The Azure AD authentication provides the possibility to use an Azure Active Directory tenant as an identity provider for Grafana. By using Azure AD Application Roles it is also possible to assign Users and Groups to Grafana roles from the Azure Portal.
+The Azure AD authentication makes it possible to use an Azure Active Directory tenant as an identity provider for Grafana. You can use Azure AD Application Roles to assign Users and Groups to Grafana roles from the Azure Portal.
```
```suggestion
The Azure AD authentication makes it possible to use an Azure Active Directory tenant as an identity provider for Grafana. You can use Azure AD Application Roles to assign users and groups to Grafana roles from the Azure Portal.
```

oddlittlebird commented on docs/sources/auth/azuread.md:
```diff
@@ -24,26 +24,31 @@ To enable the Azure AD OAuth2 you must register your application with Azure AD.
 1. Add the redirect URL `https://<grafana domain>/login/azuread`, then click **Register**.
 
-1. The app's **Overview** page is displayed. Note the **Application ID**, this is the OAuth client id.
+1. The app's **Overview** page is displayed. Note the **Application ID**: this is the OAuth client id.
```
```suggestion
1. The app's **Overview** page is displayed. Note the **Application ID**. This is the OAuth client id.
```

oddlittlebird commented on docs/sources/auth/azuread.md:
```diff
@@ -24,26 +24,31 @@ To enable the Azure AD OAuth2 you must register your application with Azure AD.
-   - Every role requires a unique id. On Linux, this can be created with `uuidgen`. For example:
+   
+   - Define the required Application Role values for Grafana: Viewer, Editor, Admin. Without this configuration, all users receive the Viewer role.
+   - Every role requires a unique id.
```
Is there any reason not to capitalize ID? If not, then please capitalize it throughout.

oddlittlebird commented on docs/sources/auth/azuread.md:
```diff
@@ -110,23 +115,28 @@ allowed_groups =
 
 > Note: Ensure that the [root_url]({{< relref "../administration/configuration/#root-url" >}}) in Grafana is set in your Azure Application Reply URLs (App -> Settings -> Reply URLs)
```
```suggestion
> **Note:** Ensure that the [root_url]({{< relref "../administration/configuration/#root-url" >}}) in Grafana is set in your Azure Application Reply URLs (App -> Settings -> Reply URLs)
```

oddlittlebird commented on docs/sources/auth/azuread.md:
```diff
@@ -110,23 +115,28 @@ allowed_groups =
-to a comma- or space-separated list of group Object Ids. Object Id for a specific group can be found on the Azure portal: go to Azure Active Directory -> Groups. For instance, if you want to
-only give access to members of the group `example` which has Id `8bab1c86-8fba-33e5-2089-1d1c80ec267d`, set
+To limit access to authenticated users who are members of one or more groups, set `allowed_groups`
+to a comma- or space-separated list of group Object Ids. Object Ids for a specific group appear on the Azure portal:
```
The capitalization of "Object Ids" is confusing. Is this in an Azure UI somewhere? If not, then "object IDs".

oddlittlebird commented on docs/sources/auth/azuread.md:
```diff
@@ -110,23 +115,28 @@ allowed_groups =
+To limit access to authenticated users who are members of one or more groups, set `allowed_groups`
+to a comma- or space-separated list of group Object Ids. Object Ids for a specific group appear on the Azure portal:
+
+1. Go to Azure Active Directory -> Groups.
```
```suggestion
1. Go to **Azure Active Directory -> Groups**.
```

oddlittlebird commented on docs/sources/auth/azuread.md:
```diff
@@ -110,23 +115,28 @@ allowed_groups =
+
+1. Go to Azure Active Directory -> Groups.
+
+For example, if you want to only give access to members of the group `example` which has Id `8bab1c86-8fba-33e5-2089-1d1c80ec267d`, set the following:
```
```suggestion
For example, if you want to only give access to members of the group `example` which has ID `8bab1c86-8fba-33e5-2089-1d1c80ec267d`, set the following:
```

oddlittlebird commented on docs/sources/auth/azuread.md:
````diff
@@ -110,23 +115,28 @@ allowed_groups =
 ```
 
-You'll need to ensure that you've [enabled group attributes](https://docs.microsoft.com/en-us/azure/active-directory/hybrid/how-to-connect-fed-group-claims#configure-the-azure-ad-application-registration-for-group-attributes) in your Azure AD Application Registration manifest file (Azure Portal -> Azure Active Directory -> Application Registrations -> Select Application -> Manifest)
+You'll need to ensure that you've [enabled group attributes](https://docs.microsoft.com/en-us/azure/active-directory/hybrid/how-to-connect-fed-group-claims#configure-the-azure-ad-application-registration-for-group-attributes) in your Azure AD Application Registration manifest file. Navigate to Azure Portal -> Azure Active Directory -> Application Registrations -> Select Application -> Manifest, a ...
````
Per the [Documentation style guide](b/master/contribute/style-guides/documentation-style-guide.md) and [Microsoft style guide](https://docs.microsoft.com/en-us/style-guide/welcome/), UI fields and things you click in the UI should be bold.

oddlittlebird reviewed (changes requested):
Great edits! I added a few suggestions.

RobinJoe committed:
Update docs/sources/auth/azuread.md

fix capital letter U and G.

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

RobinJoe committed:
Update docs/sources/auth/azuread.md

Adjust punctuation for clarity.

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

RobinJoe committed:
Update docs/sources/auth/azuread.md

Place Note in correct bold markup following style guide.

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

RobinJoe committed:
Update docs/sources/auth/azuread.md

Bold markup for UI fields.

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

RobinJoe committed:
Including suggestions as noted by Diana Payton

* adjusting object IDs to match the docs.microsoft style
* using bold for UI elements
* Other fixes as needed

Update docs/sources/auth/azuread.md

Capital for id > ID.

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

RobinJoe committed:
Update docs/sources/auth/azuread.md

Capital for id > ID.

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

RobinJoe committed:
Merge branch 'JR-PowershellNote-0209' of github.com:RobinJoe/grafana into JR-PowershellNote-0209

oddlittlebird commented on docs/sources/auth/azuread.md:
```diff
@@ -24,26 +24,31 @@ To enable the Azure AD OAuth2 you must register your application with Azure AD.
 1. Add the redirect URL `https://<grafana domain>/login/azuread`, then click **Register**.
 
-1. The app's **Overview** page is displayed. Note the **Application ID**, this is the OAuth client id.
+1. The app's **Overview** page is displayed. Note the **Application ID**. This is the OAuth client id.
```
Should it be "client ID"?

oddlittlebird replied:
If yes, then capitalize "ID" throughout.

oddlittlebird commented on docs/sources/auth/azuread.md:
```diff
@@ -24,26 +24,31 @@ To enable the Azure AD OAuth2 you must register your application with Azure AD.
-1. Click **Certificates & secrets** and add a new entry under Client secrets.
-    - Description: Grafana OAuth
-    - Expires: Never
+1. Click **Certificates & secrets** and add a new entry under **Client secrets**.
```
```suggestion
1. Click **Certificates & secrets** and then add a new entry under **Client secrets**.
```

oddlittlebird reviewed (changes requested):
Nice edits! Just a few suggestions.

RobinJoe committed:
Update docs/sources/auth/azuread.md

Add the adverb to place the next step directly after the previous step in sequence.

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

RobinJoe committed:
Update docs/sources/auth/azuread.md

Change remaining lowercase "id" use to upper case "ID" when it
appears in text.

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

RobinJoe committed:
Merge branch 'JR-PowershellNote-0209' of github.com:RobinJoe/grafana into JR-PowershellNote-0209

RobinJoe committed:
Merge branch 'master' into JR-PowershellNote-0209

achatterjee-grafana commented on docs/sources/auth/azuread.md:
```diff
@@ -14,7 +14,7 @@ weight = 700
 > Only available in Grafana v6.7+
 
-The Azure AD authentication provides the possibility to use an Azure Active Directory tenant as an identity provider for Grafana. By using Azure AD Application Roles it is also possible to assign Users and Groups to Grafana roles from the Azure Portal.
+The Azure AD authentication makes it possible to use an Azure Active Directory tenant as an identity provider for Grafana. You can use Azure AD Application Roles to assign users and groups to Grafana roles from the Azure Portal.
```
```suggestion
The Azure AD authentication allows you to use an Azure Active Directory tenant as an identity provider for Grafana. You can use Azure AD Application Roles to assign users and groups to Grafana roles from the Azure Portal.
```

achatterjee-grafana commented on docs/sources/auth/azuread.md:
```diff
@@ -24,26 +24,31 @@ To enable the Azure AD OAuth2 you must register your application with Azure AD.
-   - Add definitions for the required Application Roles for Grafana (Viewer, Editor, Admin). Without this configuration, all users will be assigned the Viewer role.
-   - Every role requires a unique id. On Linux, this can be created with `uuidgen`. For example:
+
+   - Define the required Application Role values for Grafana: Viewer, Editor, Admin. Without this configuration, all users receive the Viewer role.
```
```suggestion
   - Define the required Application Role values for Grafana: Viewer, Editor, Admin. Otherwise, all users will have the Viewer role.
```

achatterjee-grafana commented on docs/sources/auth/azuread.md:
```diff
@@ -108,25 +113,30 @@ allowed_domains =
-The `allowed_domains` option limits access to the users belonging to the specific domains. Domains should be separated by space or comma.
+### Configure allowed domains
+
+The `allowed_domains` option limits access to the users who belong to the specific domains. Separate domains with a space or comma.
```
```suggestion
The `allowed_domains` option limits access to users who belong to specific domains. Separate domains with space or comma.
```

achatterjee-grafana commented on docs/sources/auth/azuread.md:
```diff
@@ -88,7 +93,7 @@ To enable the Azure AD OAuth2 you must register your application with Azure AD.
 1. Go to **Azure Active Directory** and then to **Enterprise Applications**. Search for your application and click on it.
 
-1. Click on **Users and Groups** and add Users/Groups to the Grafana roles by using **Add User**. 
+1. Click on **Users and Groups** and add Users/Groups to the Grafana roles with **Add User**.
```
```suggestion
1. Click **Users and Groups** and add Users/Groups to the Grafana roles with **Add User**.
```

achatterjee-grafana reviewed (changes requested):
Added some copy edit suggestions.

RobinJoe commented:
Thanks for the reviews, I will continue working on the suggestions.

oddlittlebird commented:
Hey @RobinJoe, are you planning to finish this?

RobinJoe committed:
Applying achatterjee's suggestions

Accepting the changes:

* Adding in articles
* Adjusting prepositions
* Other changes as requests

Completing this commit through the Github web interface for added speed.

Co-authored-by: achatterjee-grafana <70489351+achatterjee-grafana@users.noreply.github.com>

RobinJoe committed:
Merge branch 'master' into JR-PowershellNote-0209

daniellee reviewed (approved):
Thanks @RobinJoe for going through this review process and it looks like you have applied all the feedback that you got so looks good to me.

LGRM
